apps/homescreen/test/unit/app_test.js

```diff
@@ -1,36 +1,564 @@
+      app.icons.appendChild(getIcon('gih'));
+      app.storeAppOrder();
+      assert.deepEqual(app.metadata._data, [
+        { id: 'abc/', order: 0 },
+        { id: 'def/', order: 1 },
+        { id: 'gih/', order: 2 }
+      ]);
+    });
+  });
+
+  suite('App#snapScrollPosition()', () => {
+    test('should do nothing when there are no icons', () => {
+      app.snapScrollPosition(0);
+      assert.equal(app.scrollable.style.scrollSnapPointsY, '');
+    });
+
+    suite('with icons', () => {
+      setup(() => {
+        app.icons.appendChild(getIcon('abc'));
+        sinon.stub(app.icons.firstElementChild,
+          'getBoundingClientRect', () => {
+            return { height: 100 };
+          });
+        app.scrollable = {
+          clientHeight: 200,
+          style: {},
+          scrollTop: 0,
+          scrollTo: () => {
+          }
+        };
+      });
+
+      test('should snap to icon height when the screen is small', () => {
+        app.scrollable.clientHeight = 150;
+        app.snapScrollPosition(0);
+        assert.equal(app.scrollable.style.scrollSnapPointsY, 'repeat(100px)');
+      });
+
+      test('should snap to screen height if high enough', () => {
+        app.scrollable.clientHeight = 400;
+        app.snapScrollPosition(0);
+        assert.equal(app.scrollable.style.scrollSnapPointsY, 'repeat(400px)');
+      });
+
+      test('should scroll to snap if required', () => {
+        var scrollToSy = sinon.spy(app.scrollable, 'scrollTo');
+        app.scrollable.scrollTop = 0;
+        app.snapScrollPosition(0);
+        assert.isFalse(scrollToSy.called);
+
+        app.scrollable.scrollTop = 50;
+        app.snapScrollPosition(0);
+        assert.isTrue(scrollToSy.called);
+      });
+    });
+  });
+
+  suite('App#showActionDialog()', () => {
+    var clock;
+
+    setup(function() {
+      clock = sinon.useFakeTimers();
+    });
+
+    teardown(function() {
+      clock.restore();
+    });
+
+    test('should attach events on click', () => {
+      var dialog = app.settingsDialog;
+      dialog.open = () => {
+      };
+      app.showActionDialog(dialog, null, [() => {
+      }]);
+      assert.ok(dialog.querySelector('button').onclick);
+    });
+
+    test('should set the title of the dialog', () => {
+      var dialog = app.cancelDownload;
+      dialog.open = () => {
+      };
+      assert.isFalse(dialog.querySelector('.body')
+        .hasAttribute('data-l10n-args'));
+      app.showActionDialog(dialog, 'abc', []);
+      assert.isTrue(dialog.querySelector('.body')
+        .hasAttribute('data-l10n-args'));
+      clock.tick(100);
+    });
+
+    test('should show the dialog after a little while', done => {
+      var dialog = {
+        style: { display: 'none' },
+        getElementsByClassName: () => [],
+        querySelector: () => {
+        },
+        open: () => {
+          done();
+        }
+      };
+
+      app.showActionDialog(dialog, null, []);
+      clock.tick(100);
```

Comment:
I'm using a very high value (`clock.tick(10000);`) to avoid exporting the real one here.